packages/@aws-cdk/assets/lib/asset.ts

```diff
@@ -121,12 +137,20 @@ export class Asset extends cdk.Construct {
     // form the s3 URL of the object key
     this.s3Url = this.bucket.urlForObject(this.s3ObjectKey);
 
+    this.artifactName = this.node.uniqueId;
+
+    this.relativePath = props.packaging === AssetPackaging.ZipDirectory
+      ? this.artifactName
+      : path.join(this.artifactName, path.basename(props.path));
+
+    this.relativePath = `./${this.relativePath}`; // prefix with './'
```

Comment:
Why prefix with `./`? By traditional filesystem convention, any path that doesn't start with `/` is already relative.

If you're thinking about NodeJS, they reason they do it is because they also want to support a global registry of packages, and they want to disambiguate:

```ts
require('bla');   // Is this a global 'bla' or a local 'bla.js'?
```

I don't think we need that feature, and even if we did we could get it by prepending a special namespace like `builtin:bla`.